### PR TITLE
Adding Pymem3dg to conda-forge

### PR DESCRIPTION
**Pull request by ctlee**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by cuzhucuncheng** [comment]
I confirm that I am willing to be a maintainer of Mem3DG.

**Comment by dopplershift** [comment]
Need to revert the changes to files in `recipes/libdc1394/`--you shouldn't touch any files in other directories.

Need to also figure out what's going wrong on linux/osx.

**Comment by ctlee** [comment]
@dopplershift thanks for the notes! Things appear to be working as expected -- CI is passing. Please let me know if there are any other fixes which should be made.

File: recipes/pymem3dg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    - xorg-libx11  # [linux]
+    - xorg-libxrandr  # [linux]
+    - xorg-libxinerama  # [linux]
+    - xorg-libxcursor  # [linux]
+    - xorg-libxi  # [linux]
+    - importlib_resources  # [py<39]
+  run:
+    - python
+    - netcdf-cxx4
+    - xorg-libx11  # [linux]
```
[the remark this answers is not in the document]
Comment (ctlee) [reply recipes/pymem3dg/meta.yaml:47]: We're currently using [polyscope](https://polyscope.run/) to handle the visualization of the trajectories in Mem3DG. I'm aware that there's a [polyscope feedstock](https://github.com/conda-forge/polyscope-feedstock) but afaik it doesn't bundle a linkable shared library and headers. We're using the c++ api to add options to the visualization so the python library isn't enough.

In the long run, I've been meaning to talk to @nmwsharp about packaging development versions of polyscope and geometry-central on conda-forge. For now, they're bundled in this application with licenses etc. until that happens.

**Comment by ctlee** [comment]
@conda-forge-admin, please ping team

Are there any other modifications needed for this contribution?

File: recipes/pymem3dg/meta.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmwsharp) [reply recipes/pymem3dg/meta.yaml:47]: I would be to have polyscope/geometry-central packaged on conda forge! And in the meantime bundling them like here is of course fine.

I'm still a conda forge novice---just barely figured out how to get that polyscope python feedstock up :)  However if you (or anyone else) can provide some pointers or submit a pull I'm all for it.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**